synapse/api/ratelimiting.py

```diff
@@ -61,12 +61,16 @@ class Ratelimiter:
     """
 
     def __init__(
-        self, store: DataStore, clock: Clock, rate_hz: float, burst_count: int
+        self,
+        store: DataStore,
+        clock: Clock,
+        cfg: RatelimitSettings,
     ):
         self.clock = clock
-        self.rate_hz = rate_hz
-        self.burst_count = burst_count
+        self.rate_hz = cfg.per_second
+        self.burst_count = cfg.burst_count
         self.store = store
+        self._limiter_name = cfg.key
```

Comment:
I contemplated just storing `self.cfg` here, but went with this approach to try and keep the diff smaller.

Reply:
I think *not* storing the config is a bit simpler when you're looking at later code. (less indirection). tl;dr -- I would leave it as is.

Reply:
On the flip side, that indirection makes it a little easier to reload config on SIGHUP. Definitely not a conversation for now, though.